Lemonize New insight button

Twixes opened the pull request:
## Changes

Reworked the New insight button to the redesign's system. Also just fixed it.

| Before (??) | After |
| --- | --- |
| ![before](https://user-images.githubusercontent.com/4550621/141375933-a4d63aa4-5852-4beb-b576-5bae0cbc3e2a.png) | ![after](https://user-images.githubusercontent.com/4550621/141375928-ed496fe5-48d8-4906-8831-c712c968e4e4.png) |

Twixes committed:
Lemonize New insight button

clarkus reviewed (approved):
Looks good to me. My same shadow feedback from the other issue still applies, but we can figure that out in the other PR.

paolodamico commented:
Quick feedback, not a huge fan of the box not having any spacing between the button and the top of the box, but I'll defer to @clarkus

clarkus commented:
> Quick feedback, not a huge fan of the box not having any spacing between the button and the top of the box, but I'll defer to @clarkus

Oh yes good catch. Those are usually placed 4–8px from the edge of the button depending on context.

mariusandra commented on frontend/src/lib/components/LemonButton/LemonButton.scss:
```diff
@@ -1,3 +1,4 @@
+@use "sass:color";
```
Not a fan of this. Every extra thing we import 1) slows down builds, 2) complicates builds by requiring even more custom plugins, 3) makes it harder to migrate to any other CSS framework down the line. 

Let's defer as much as we can to CSS, instead of the build scripts. Hence, could you either use the CSS brightness filter, or then hardcode the `--primary-light` and `--primary-lighter` vars so that we can reuse them instead of hoping that the `-5` and `-10` lightness pattern will be respected everywhere else.

mariusandra commented on frontend/src/scenes/saved-insights/SavedInsights.tsx:
```diff
@@ -120,6 +121,52 @@ const columnSort = (direction: 'up' | 'down' | 'none'): JSX.Element => (
 )
 
+function NewInsightButton(): JSX.Element {
+    const [isNewInsightsPopupVisible, setIsNewInsightsPopupVisible] = useState(false)
```
:'(

Twixes committed:
Address feedback

Twixes commented:
Addressed feedback:
- hardcoded colors as CSS vars (this is actually also better in that I copied variations of `--primary` calculated by AntD, so that hover/active states look the same for Lemon and AntD components)
- added a 4px gap between the `Popup` overlay and its reference element

Twixes commented:
Demo:
![Zrzut ekranu 2021-11-12 o 17 25 40](https://user-images.githubusercontent.com/4550621/141500770-95fc3f7a-5c66-4f8a-869d-bcd6a37bf25f.png)

mariusandra reviewed (approved):
Overall looks good, though some of these icons feel a bit off, yet I don't have suggestions for improvements.
![image](https://user-images.githubusercontent.com/53387/141503147-9c36a96f-1aa1-492e-aee6-10c82127eaea.png)

Twixes commented:
What can be improved about the icons? Happy to do a follow-up PR about that if needed
